packages/next/client/dev-build-watcher.js

```diff
@@ -0,0 +1,205 @@
+import { getEventSourceWrapper } from './dev-error-overlay/eventsource'
+
+export default function initializeBuildWatcher () {
+  const shadowHost = document.getElementById('__next-build-watcher')
+  if (!shadowHost) return
+  let shadowRoot
+  let prefix = ''
+
+  if (shadowHost.attachShadow) {
+    shadowRoot = shadowHost.attachShadow({ mode: 'open' })
+  } else {
+    // If attachShadow is undefined then the browser does not support
+    // the Shadow DOM, we need to prefix all the names so there
+    // will be no conflicts
+    shadowRoot = shadowHost
+    prefix = '__next-build-watcher-'
+  }
+
+  // Get the current style
+  const style = shadowHost.getAttribute('data-buildwatcherstyle')
+
+  // Container
+  const container = createContainer(prefix)
+  shadowRoot.appendChild(container)
+
+  // CSS
+  const css = createCss(style, prefix)
+  shadowRoot.appendChild(css)
+
+  // State
+  let isVisible = false
+  let isBuilding = false
+  let timeoutId = null
+
+  // Handle events
+  const evtSource = getEventSourceWrapper({ path: '/_next/webpack-hmr', ondemand: 1 })
```

Comment:
Now using `getEventSourceWrapper` due to a27c2352600d428b848bc70fb59374b3f8f50f8e. I am not sure if the `path` will always be this one that I included.